src/charm_state.py

```diff
@@ -31,6 +35,19 @@ class ARCH(str, Enum):
 COS_AGENT_INTEGRATION_NAME = "cos-agent"
 
 
+class RunnerStorage(str, Enum):
+    """Supported storage as runner disk."""
+
+    JUJU_STORAGE = "juju-storage"
+    MEMORY = "memory"
+
+    @classmethod
+    def _missing_(cls, value: object) -> Any:
+        raise ValueError(
+            f"{value} is not a valid runner storage ({list(cls.__members__.values())})"
+        )
```

Comment:
I'm curios around this one, does this customise the error message? I'm guessing by default the enum wouldn't allow a value that isn't defined on it to be selected?

Reply:
This does customise the error message. 
Enum does checks for non-valid values, but the error message would be like: "X is not a valid RunnerStorage".

As oppose to:
"X is not a valid runner-storage (memory, juju-storage)"

I am fine either way. 
`runner-storage` is what the configuration is called, but I am guessing most user could infer that RunnerStorage is referring to `runner-storage`.

Reply:
I end up moving the custom error message up the stack.